### ISSUE TITLE
New articles aren't getting displayed

### ISSUE THREAD

**Issue by fb87** [jangernert/FeedReader#168]
I received new articles as can be seen in the feed pane, but they aren't displayed in the news pane until I restart the program.

![feedreader_012](https://cloud.githubusercontent.com/assets/10520767/17277824/43e79242-574d-11e6-8dca-6248e41400b8.png)
[feedreader.tar.gz](https://github.com/jangernert/FeedReader/files/392514/feedreader.tar.gz)

**Comment by jangernert** [comment]
And if you click on another feed and then back on to "All Articles" ?

**Comment by fb87** [comment]
Didn't change anything (but I haven't tried hitting „all“ or „marked“ at the top).

**Comment by jangernert** [comment]
Is the problem reproducible or has it happened only once?

**Comment by fb87** [comment]
Just this time

**Comment by jangernert** [comment]
=/

**Comment by luislago** [comment]
Getting the same issue on Fedora, with my Feedly account

**Comment by mekeor** [comment]
Same issue here.

First, I used the latest dev-version on GitHub (which I built from scratch). After it didn't work, I installed the stable version from scratch. It still didn't work, that is, it didn't fetch/show my feeds. But after restarting feedreade like ~10 times, now it worked.

These are the messages being written by `feedreader --debug` over and over again while trying to update my feeds.

```
[ DEBUG ] ArticleList: newList
[ DEBUG ] ArticleList: create
[ DEBUG ] ArticleList: offset 0
[ DEBUG ] limit: 20
[ DEBUG ] load articles from db
[ DEBUG ] SELECT ROWID, feedID, articleID, title, author, url, preview, unread, marked, tags, date, guidHash FROM main.articles ORDER BY rowid DESC LIMIT 20 OFFSET 0
[ DEBUG ] actual articles loaded: 0
[ DEBUG ] ArticleList: insert new rows
[ DEBUG ] ArticleList: create finished
[ DEBUG ] DBusConnection: writeInterfaceState
[ DEBUG ] ArticleList: get State
[ DEBUG ] scrollpos 0,000000
[ DEBUG ] scrollpos 0,000000
[ DEBUG ] offset 0
[ DEBUG ] DBusConnection: updateFeedList
[ DEBUG ] DBusConnection: updateArticleList
[ DEBUG ] ArticleList: updateArticleList
[WARNING] ArticleList: updateArticleList - list was empty, so no reason to update - will launch newList()
```

I'm running Debian testing and using Feedly.

**Comment by jangernert** [comment]
@mekeor
So didn't it fetch your feeds or didn't it show your articles? The two things are very different.
Are you sure you are experiencing the same issue? Was the sidebar populated by your categories and feeds which had the proper unread counters on them?
Also, why are you starting feedreader with the "--debug" argument? This mode is NOT what you think it is.
Sorry, but this comment is more confusing than helpful.

**Comment by fb87** [comment]
Now I had a little different issue – an wild artifact appeared on the top of the news list as you can see in the screenshot. So I thought I switch from all feeds to a specific one and back – and the feed list should be loaded new as a whole and the artifact would be gone. But switching to another feed had no influence on the news list, so everything there stayed like what it was with „all feeds“ marked. Logs are attached.

![feedreader_002](https://cloud.githubusercontent.com/assets/10520767/18565447/17ce4478-7b90-11e6-935b-7683d36b1b63.png)

[feedreader.tar.gz](https://github.com/jangernert/FeedReader/files/475474/feedreader.tar.gz)

**Comment by jangernert** [comment]
I really like to know how you do this stuff...
That's a different issue though. The article list never finished adding new rows (I guess the bar up top are actually lots of bunched up article-rows that are mid reveal-transition). And now the application is patiently waiting for the article-list to finish up it's last action before a new one can be started. But it never finishes up.
So unless you can tell me exactly how to reproduce this issue, I can't really do anything about it.

**Comment by fb87** [comment]
The only thing other than reading news I did, was pushing the update icon, because it didn't update upon waking up from standby. ;)

**Comment by skewty** [comment]
I am a first time new install and as such don't have any articles.  I believe I am having the same issue though:

```
[ DEBUG ] offset 0
[ DEBUG ] DBusConnection: updateFeedList
[ DEBUG ] DBusConnection: updateArticleList
[ DEBUG ] ArticleList: updateArticleList
[WARNING] ArticleList: updateArticleList - list was empty, so no reason to update - will launch newList()
[ DEBUG ] ArticleList: newList
[ DEBUG ] ArticleList: create
[ DEBUG ] ArticleList: offset 0
[ DEBUG ] limit: 20
[ DEBUG ] load articles from db
[ DEBUG ] SELECT ROWID, feedID, articleID, title, author, url, preview, unread, marked, tags, date, guidHash FROM main.articles WHERE feedID IN ("feed/https://communities.intel.com/community/feeds/messages?thread=77934") AND unread = 9  ORDER BY rowid DESC LIMIT 20 OFFSET 0
[ DEBUG ] actual articles loaded: 0
[ DEBUG ] ArticleList: insert new rows
[ DEBUG ] ArticleList: create finished
[ DEBUG ] DBusConnection: writeInterfaceState
[ DEBUG ] ArticleList: get State
[ DEBUG ] scrollpos 0.000000
[ DEBUG ] scrollpos 0.000000
[ DEBUG ] offset 0
[ DEBUG ] DBusConnection: updateFeedList
[ DEBUG ] DBusConnection: updateArticleList
[ DEBUG ] ArticleList: updateArticleList
[WARNING] ArticleList: updateArticleList - list was empty, so no reason to update - will launch newList()
[ DEBUG ] ArticleList: newList
[ DEBUG ] ArticleList: create
[ DEBUG ] ArticleList: offset 0
[ DEBUG ] limit: 20
[ DEBUG ] load articles from db
[ DEBUG ] SELECT ROWID, feedID, articleID, title, author, url, preview, unread, marked, tags, date, guidHash FROM main.articles WHERE feedID IN ("feed/https://communities.intel.com/community/feeds/messages?thread=77934") AND unread = 9  ORDER BY rowid DESC LIMIT 20 OFFSET 0
[ DEBUG ] actual articles loaded: 0
[ DEBUG ] ArticleList: insert new rows
[ DEBUG ] ArticleList: create finished
[ DEBUG ] DBusConnection: writeInterfaceState
[ DEBUG ] ArticleList: get State
[ DEBUG ] scrollpos 0.000000
[ DEBUG ] scrollpos 0.000000
[ DEBUG ] offset 0
[ DEBUG ] DBusConnection: updateFeedList
[ DEBUG ] DBusConnection: updateArticleList
[ DEBUG ] ArticleList: updateArticleList
[WARNING] ArticleList: updateArticleList - list was empty, so no reason to update - will launch newList()
[ DEBUG ] ArticleList: newList
[ DEBUG ] ArticleList: create
[ DEBUG ] ArticleList: offset 0
[ DEBUG ] limit: 20
[ DEBUG ] load articles from db
[ DEBUG ] SELECT ROWID, feedID, articleID, title, author, url, preview, unread, marked, tags, date, guidHash FROM main.articles WHERE feedID IN ("feed/https://communities.intel.com/community/feeds/messages?thread=77934") AND unread = 9  ORDER BY rowid DESC LIMIT 20 OFFSET 0
[ DEBUG ] actual articles loaded: 0
[ DEBUG ] ArticleList: insert new rows
[ DEBUG ] ArticleList: create finished
```

**Comment by jangernert** [comment]
You just don't seem to have any articles in your db that belong to the feed you are viewing. Can't display what isn't there ;)

**Comment by jangernert** [comment]
Since the ArticleList-code has been mostly rewritten, I'll close this. 
If something similar happens again, please open up a new issue.

**Comment by fb87** [comment]
Now when new news are fetched, (at least) the "all articles" category doesn't get refreshed until I select another feed and swap back.

**Comment by fbruetting** [comment]
Still an issue. Oftentimes the list doesn't contain all the articles indicated in the new article number on the right side of the feed list. I have to switch to another feed and re-switch to the former feed to see the former not contained articles.
